packages/style-value-parser/src/at-queries-from-tokens/media-query.js

```diff
@@ -277,35 +295,103 @@ const mediaOrRulesParser: TokenParser<MediaOrRules | MediaQueryRule> =
+    // new branch: allow a media keyword only if enclosed in parentheses
+    () =>
+      basicMediaTypeParser
+        .surroundedBy(
+          TokenParser.tokens.OpenParen,
+          TokenParser.tokens.CloseParen,
+        )
+        .map((keyword) => ({
+          type: 'media-keyword',
+          key: keyword,
+          not: false,
+        })),
+    mediaAndRulesParser,
+    mediaOrRulesParser,
+    simplePairParser,
+    mediaWordRuleParser,
+    () => notParser,
+    () =>
+      mediaOrRulesParser.surroundedBy(
+        TokenParser.tokens.OpenParen,
+        TokenParser.tokens.CloseParen,
+      ),
+    () =>
+      mediaAndRulesParser.surroundedBy(
+        TokenParser.tokens.OpenParen,
+        TokenParser.tokens.CloseParen,
+      ),
+  );
+
+// now define notParser using the helper
+notParser = TokenParser.sequence(
+  TokenParser.tokens.OpenParen,
+  TokenParser.string('not'),
+  TokenParser.tokens.Whitespace,
+  getNormalRuleParser(),
+  TokenParser.tokens.CloseParen,
+).map(([_openParen, _not, _space, rule, _closeParen]) => ({
+  type: 'not',
+  rule,
+}));
+
 export class MediaQuery {
   queries: MediaQueryRule;
-  constructor(queries: this['queries']) {
+  constructor(queries: MediaQueryRule) {
     this.queries = MediaQuery.normalize(queries);
   }
   toString(): string {
     return `@media ${this.#toString(this.queries, true)}`;
   }
   #toString(queries: MediaQueryRule, isTopLevel: boolean = false): string {
     switch (queries.type) {
-      case 'media-keyword':
-        return queries.not ? `not ${queries.key}` : queries.key;
+      case 'media-keyword': {
+        const prefix = queries.not ? 'not ' : queries.only ? 'only ' : '';
+        return prefix + queries.key;
+      }
       case 'word-rule':
         return `(${queries.keyValue})`;
       case 'pair': {
-        const valueStr = Array.isArray(queries.value)
-          ? `${queries.value[0]} / ${queries.value[2]}`
-          : typeof queries.value === 'string'
-            ? queries.value
-            : `${queries.value.value}${queries.value.unit}`;
-
+        let valueStr: string;
+        if (Array.isArray(queries.value)) {
+          valueStr = `${queries.value[0]} / ${queries.value[2]}`;
+        } else if (typeof queries.value === 'string') {
+          valueStr = queries.value;
+        } else if (
+          typeof queries.value === 'object' &&
+          queries.value !== null &&
+          typeof queries.value.toString === 'function'
+        ) {
+          const candidate = queries.value.toString();
+          if (
+            candidate === '[object Object]' &&
+            typeof queries.value !== 'string' &&
+            !Array.isArray(queries.value)
+          ) {
+            const dim = queries.value as Length;
+            valueStr = `${dim.value}${dim.unit}`;
+          } else {
+            valueStr = candidate;
+          }
+        } else {
+          valueStr = '';
+        }
         return `(${queries.key}: ${valueStr})`;
```

Comment:
This long chain can be simplified with early returns.

```suggestion
        if (Array.isArray(queries.value)) {
          return `(${queries.key}: ${queries.value[0]} / ${queries.value[2]})`;
        }
        if (typeof queries.value === 'string') {
          return `(${queries.key}: ${queries.value})`;
        } 
        if (
          typeof queries.value === 'object' &&
          queries.value !== null &&
          typeof queries.value.toString === 'function'
        ) {
          const candidate = queries.value.toString();
          if (
            candidate === '[object Object]' &&
            typeof queries.value !== 'string' &&
            !Array.isArray(queries.value)
          ) {
            const dim = queries.value as Length;
            return `(${queries.key}: ${dim.value}${dim.unit})`;
          }
          return `(${queries.key}: ${candidate})`;
        } 
        return `(${queries.key}: )`;
```

This reveals a problem with the last `else` case where `(${queries.key}: )` doesn't seem like a valid thing to return.

Reply:
added a throw in that case, let's support this set of values first